worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -588,25 +597,28 @@ public DiskFileInfo resolve(
               fileId,
               () -> {
                 FileChannel indexChannel = null;
-                FSDataInputStream hdfsIndexStream = null;
-                boolean isHdfs = Utils.isHdfsPath(indexFilePath);
+                FSDataInputStream dfsIndexStream = null;
+                boolean isDfs = Utils.isHdfsPath(indexFilePath) || Utils.isS3Path(indexFilePath);
+                boolean isS3 = Utils.isS3Path(indexFilePath);
                 int indexSize;
                 try {
-                  if (isHdfs) {
-                    hdfsIndexStream = StorageManager.hadoopFs().open(new Path(indexFilePath));
-                    indexSize =
-                        (int)
-                            StorageManager.hadoopFs()
-                                .getFileStatus(new Path(indexFilePath))
-                                .getLen();
+                  if (isDfs) {
```

Comment:
Do not write codes like 
https://github.com/apache/celeborn/blob/272412287d33e3701e9d93a1237901de8323a6db/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java#L605-L611

Reply:
A file sorter will handle one partition location only. Save the corresponding hadoopFs to the object will be fine.

Reply:
OK. Done